### PR TITLE
Removed deprecated React life cycle methods

### PR DESCRIPTION
**Pull request by dipiash**
First fix for #4479

### CODE REVIEW

**Review by assimelha — APPROVED** [review]
Working perfectly

**Review by timneutkens — CHANGES_REQUESTED** [review]
For all these PRs we'd need to do add the `react-lifecycles-compat` polyfill, to be compatible with older versions. I'd like to do that for the tests even.

File: test/integration/basic/pages/stateful.js [hunk 1]
```diff
@@ -8,8 +8,10 @@ export default class Statefull extends Component {
     this.state = { answer: null }
   }
 
-  componentWillMount () {
-    this.setState({ answer: 42 })
+  static getDerivedStateFromProps (nextProps, prevState) {
+    if (!prevState.answer) {
+      return {answer: 42}
```
Comment (piuccio) [inline test/integration/basic/pages/stateful.js]: why not just setting 42 in the constructor?

**Comment by timneutkens** [comment]
I believe we should remove the test for this, as it's an anti-pattern when doing SSR. I'll do it sometime in the future.

**Pull request closed without merging**